src/lib.rs

```diff
@@ -132,6 +140,55 @@ impl File {
         }
     }
 
+    /// Creates a new [`File`] from an existing file on disk.
+    ///
+    /// # Arguments
+    /// - `path` - The path to the file.
+    ///
+    /// # Returns
+    /// - [`File`] - The new File.
+    ///
+    /// # Example
+    /// ```
+    /// let file = piston_rs::File::load_from("src/lib.rs");
+    ///
+    /// assert!(file.content.contains("pub fn load_from"));
+    /// assert_eq!(file.name, "lib.rs".to_string());
+    /// assert_eq!(file.encoding, "utf8".to_string());
+    /// ```
+    pub fn load_from(path: &str) -> Self {
+        let path = PathBuf::from(path);
+
+        if !path.is_file() {
+            println!("File does not exist, or is a directory -- using defaults");
+            return File::default();
+        }
+
+        let name = match path.file_name() {
+            Some(n) => n.to_str().unwrap_or("Invalid file name"),
+            None => {
+                println!("Invalid file name");
+                ""
+            }
+        };
+
+        Self {
+            name: name.to_string(),
+            content: File::load_contents(&path),
+            encoding: String::from("utf8"),
+        }
+    }
+
+    fn load_contents(path: &Path) -> String {
+        match fs::read_to_string(&path) {
+            Ok(c) => c,
+            Err(e) => {
+                println!("{}", e);
+                String::new()
+            }
+        }
+    }
```

Comment:
Here propagating the Result is much easier

```suggestion
    fn load_contents(path: &Path) -> Result<String, Box<dyn error::Error>> {
        Ok(fs::read_to_string(&path)?)
    }
```